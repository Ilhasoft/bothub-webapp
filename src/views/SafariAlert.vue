<template>
  <div class="safari-alert">
    <div class="safari-alert__content">
      <img
        src="@/assets/imgs/botinho.png"
        alt="botinho">
      <h3 class="safari-alert__content__title">
        Sorry, the Safari browser is not compatible with all our features.
      </h3>
      <h3 class="safari-alert__content__title">
        We strongly recommend that you switch to a different browser.
      </h3>
      <div class="bh-grid">
        <a
          class="bh-grid__item"
          href="https://www.google.com/intl/pt-BR_ALL/chrome/">
          <img
            src="@/assets/imgs/chrome.svg"
            alt="chrome"
            class="safari-alert__content__browser-icon">
        </a>
        <a
          class="bh-grid__item"
          href="https://www.mozilla.org/pt-BR/firefox/new/">
          <img
            src="@/assets/imgs/mozila.svg"
            alt="mozila"
            class="safari-alert__content__browser-icon">
        </a>
        <a
          class="bh-grid__item"
          href="https://www.opera.com/pt-br/download">
          <img
            src="@/assets/imgs/opera.svg"
            alt="opera"
            class="safari-alert__content__browser-icon">
        </a>
      </div>
      <bh-button
        class="safari-alert__content__button"
        size="medium"
        primary
        @click="goToLandingPage()">Keep using Safari</bh-button>
    </div>
  </div>
</template>

<script>

export default {
  name: 'SafariAlert',
  beforeMount() {
    const chrome = navigator.userAgent.indexOf('Chrome') > -1;
    const safari = navigator.userAgent.indexOf('Safari') > -1;

    if (!safari || chrome) {
      this.$router.push({
        name: 'landingPage',
      });
    }
  },
  methods: {
    goToLandingPage() {
      this.$router.push({
        name: 'landingPage',
      });
    },
  },
};
</script>

<style lang="scss" scoped>
@import '~bh/src/assets/scss/colors.scss';
@import '~@/assets/scss/utilities.scss';


.safari-alert{
  background-color: $color-white;

  &__content {
    background-color: $color-white;
    margin: 4rem auto;
    text-align: center;
    max-width: $max-repository-card-width - 300px;

    &__title {
      margin: 3rem;
    }

    &__browser-icon {
      height: 5rem;
      margin: 0 auto;
      cursor: pointer;
      transition: all .3s ease-in-out;

      &:hover {
        transform: scale(1.2);
      }
    }

    &__button {
      margin-top: 2rem;
    }
  }
}
</style>
