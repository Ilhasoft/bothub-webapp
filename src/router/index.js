import Vue from 'vue';
import Router from 'vue-router';
import Home from '@/views/Home';
import LandingPage from '@/views/LandingPage';
import SignUp from '@/views/auth/SignUp';
import MyProfile from '@/components/MyProfile';
import ResetPassword from '@/components/ResetPassword';
import RepositoryHome from '@/views/repository/Home';
import RepositoryTrainings from '@/views/repository/Trainings';
import RepositoryTranslate from '@/views/repository/Translate';
import RepositoryTranslations from '@/views/repository/Translations';
import RepositorySettings from '@/views/repository/Settings';
import RepositoryAnalyzeText from '@/views/repository/AnalyzeText';
import RepositoryEvaluate from '@/views/repository/Evaluate';
import RepositoryVersions from '@/views/repository/Versions';
import RepositoryLog from '@/views/repository/Log';
import NotFound from '@/views/NotFound';
import SafariAlert from '@/views/SafariAlert';
import store from '../store';

import DashboardLayout from '@/layout/dashboard/DashboardLayout';


Vue.use(Router);

export default new Router({
  mode: 'history',
  routes: [
    {
      path: '/',
      name: 'landingPage',
      component: LandingPage,
      beforeEnter: async (to, from, next) => {
        if (store.getters.authenticated) {
          next('/home');
        } else {
          next();
        }
      },
    },
    {
      path: '/signup',
      name: 'signUp',
      component: SignUp,
      beforeEnter: async (to, from, next) => {
        if (store.getters.authenticated) {
          next('/home');
        } else {
          next();
        }
      },
    },
    {
      path: '/home',
      name: 'home',
      component: Home,
    },
    {
      path: '/reset-password/:nickname/:token/',
      component: ResetPassword,
    },
    {
      path: '/myprofile/',
      name: 'myProfile',
      component: MyProfile,
      beforeEnter: async (to, from, next) => {
        if (!store.getters.authenticated) {
          store.dispatch('openLoginModal', {
            next: to,
            redirectToWhenFails: { name: 'home' },
          });
        } else {
          next();
        }
      },
    },
    {
      path: '/dashboard',
      name: 'dashboard',
      component: DashboardLayout,
      children: [
        {
          path: ':ownerNickname/:slug/',
          name: 'repository-summary',
          component: RepositoryHome,
        },
        {
          path: ':ownerNickname/:slug/training/',
          name: 'repository-training',
          component: RepositoryTrainings,
        },
        {
          path: ':ownerNickname/:slug/translate/',
          name: 'repository-translate',
          component: RepositoryTranslate,
        },
        {
          path: ':ownerNickname/:slug/translations/',
          name: 'repository-translations-status',
          component: RepositoryTranslations,
        },
        {
          path: ':ownerNickname/:slug/Integration/',
          name: 'repository-integration',
          component: RepositoryAnalyzeText,
        },
        {
          path: ':ownerNickname/:slug/settings/',
          name: 'repository-settings',
          component: RepositorySettings,
        },
        {
          path: ':ownerNickname/:slug/evaluate/',
          name: 'repository-test',
          component: RepositoryEvaluate,
        },
        {
          path: ':ownerNickname/:slug/versions/',
          name: 'repository-versions',
          component: RepositoryVersions,
        },
        {
          path: ':ownerNickname/:slug/log/',
          name: 'repository-log',
          component: RepositoryLog,
        },
      ],
    },
    {
      path: '*',
      name: '404',
      component: NotFound,
    },
    {
      path: '/safariAlert/',
      name: 'safari-alert',
      component: SafariAlert,
    },
  ],
  scrollBehavior() {
    return { x: 0, y: 0 };
  },
});
