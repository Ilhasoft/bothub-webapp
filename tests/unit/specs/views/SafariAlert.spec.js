import { shallowMount, createLocalVue } from '@vue/test-utils';

import SafariAlert from '@/views/SafariAlert';
import router from '@/router';


const localVue = createLocalVue();


describe('SafariAlert.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(SafariAlert, {
      localVue,
      mocks: {
        $t: () => 'some specific text',
      },
      router,
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
