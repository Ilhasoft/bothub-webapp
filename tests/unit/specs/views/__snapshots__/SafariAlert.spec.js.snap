// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SafariAlert.vue renders correctly 1`] = `
<div class="safari-alert">
  <div class="safari-alert__content">
    <article class="safari-alert__content__article">
      <h3 class="safari-alert__content__title">
        some specific text
      </h3>
      <p class="safari-alert__content__text">
        some specific text
      </p>
    </article>
    <div class="safari-alert__content__grid"><a href="https://www.mozilla.org/pt-BR/firefox/new/"><img src="@/assets/imgs/mozila.svg" alt="mozila" class="safari-alert__content__browser-icon">
        <p>Firefox</p>
      </a> <a href="https://www.google.com/intl/pt-BR_ALL/chrome/"><img src="@/assets/imgs/chrome.svg" alt="chrome" class="safari-alert__content__browser-icon">
        <p>Chrome</p>
      </a> <a href="https://www.opera.com/pt-br/download"><img src="@/assets/imgs/opera.svg" alt="opera" class="safari-alert__content__browser-icon">
        <p>Opera</p>
      </a></div>
    <b-button-stub size="normal" nativetype="button" tag="button" class="safari-alert__content__button">
      some specific text
    </b-button-stub>
  </div>
</div>
`;
